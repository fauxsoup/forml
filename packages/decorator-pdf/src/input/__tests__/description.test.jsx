import Description from '../description';
import Context from '@forml/context';
import React from 'react';
import ReactPDF from '@react-pdf/renderer';
import { render } from '@testing-library/react';
import * as decorator from '../../';

describe('renders', function () {
    let form;
    let title = 'title';
    let description = 'description';
    let styles;
    let descriptionStyles;

    beforeEach(function () {
        form = { type: 'fieldset', description: 'description' };

        descriptionStyles = jest.fn(() => ({ border: '1pt solid red' }));

        styles = {
            get description() {
                return descriptionStyles();
            },
        };
    });

    test('with title and description', function () {
        const { container } = render(
            <ReactPDF.Page>
                <Context.Provider value={{ decorator }}>
                    <Description
                        form={form}
                        title={title}
                        description={description}
                    >
                        {description}
                    </Description>
                </Context.Provider>
            </ReactPDF.Page>
        );

        expect(container).toMatchSnapshot();
    });

    test('with title and no description', function () {
        const { container } = render(
            <ReactPDF.Page>
                <Context.Provider value={{ decorator }}>
                    <Description form={form} description={title}>
                        {description}
                    </Description>
                </Context.Provider>
            </ReactPDF.Page>
        );

        expect(container).toMatchSnapshot();
    });

    test('with no title or description', function () {
        const { container } = render(
            <ReactPDF.Page>
                <Context.Provider value={{ decorator }}>
                    <Description form={form}>{description}</Description>
                </Context.Provider>
            </ReactPDF.Page>
        );

        expect(container).toMatchSnapshot();
    });

    test('with relevant formStyles', function () {
        form = { ...form, styles };
        const { container } = render(
            <ReactPDF.Page>
                <Context.Provider value={{ decorator }}>
                    <Description form={form}>{description}</Description>
                </Context.Provider>
            </ReactPDF.Page>
        );

        expect(container).toMatchSnapshot();
        expect(descriptionStyles).toHaveBeenCalled();
    });
});
