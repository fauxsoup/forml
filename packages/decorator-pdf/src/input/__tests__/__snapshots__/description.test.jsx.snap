// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with no title or description 1`] = `
<div>
  <page>
    <view
      style="margin-top: 2pt; font-size: 9pt; color: rgba(0, 0, 0, 0.54);"
    >
      <text>
        description
      </text>
    </view>
  </page>
</div>
`;

exports[`renders with relevant formStyles 1`] = `
<div>
  <page>
    <view
      style="margin-top: 2pt; font-size: 9pt; color: rgba(0, 0, 0, 0.54); border: 1pt solid red;"
    >
      <text>
        description
      </text>
    </view>
  </page>
</div>
`;

exports[`renders with title and description 1`] = `
<div>
  <page>
    <view
      style="margin-top: 2pt; font-size: 9pt; color: rgba(0, 0, 0, 0.54);"
    >
      <text>
        description
      </text>
    </view>
  </page>
</div>
`;

exports[`renders with title and no description 1`] = `
<div>
  <page>
    <view
      style="margin-top: 2pt; font-size: 9pt; color: rgba(0, 0, 0, 0.54);"
    >
      <text>
        description
      </text>
    </view>
  </page>
</div>
`;
